alerter/Alerter.java

```diff
@@ -0,0 +1,64 @@
+class Alerter {
+    static int alertFailureCount = 0;
+
+    static final int THRESHOLD_VALUE = 200;
+
+    static final String ENV_TYPE = "DEVELOPMENT";
+
+    enum EnvType {
+        DEVELOPMENT, PRODUCTION
+    }
+
+    static void alertInCelsius(float fahrenheit) {
+        NetworkAlert networkAlert = getNetworkAlert();
+        float celsius = (fahrenheit - 32) * 5 / 9;
+        int returnCode = networkAlert.alert(celsius, THRESHOLD_VALUE);
+        if (returnCode != 200) {
+            alertFailureCount += 1;
+        }
+    }
+
+    private static NetworkAlert getNetworkAlert() {
+        NetworkAlert networkAlert = null;
+        if (EnvType.DEVELOPMENT.name().equals(ENV_TYPE)) {
+            networkAlert = new NetworkAlertStubImpl();
+        } else if (EnvType.PRODUCTION.name().equals(ENV_TYPE)) {
+            networkAlert = new NetworkAlertImpl();
+        }
+        return networkAlert;
+    }
+
+    public static void main(String[] args) {
+        alertInCelsius(400.5f);
+        alertInCelsius(303.6f);
+        alertInCelsius(500.08f);
+        System.out.printf("%d alerts failed.\n", alertFailureCount);
+        assert (alertFailureCount == 2);
+    }
+}
+
+interface NetworkAlert {

```

Comment:
you have defined an interface. 50% of the problem solved 👍 
But the creation of the object is not at the right place. Instead of if/else based on envt type in production code, the object should have been created in the main and then passed to production code.

Also the test code and production code should be in separate files.